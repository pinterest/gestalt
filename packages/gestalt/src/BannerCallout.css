.dismissButton {
  position: absolute;
}

html[dir="rtl"] .rtlPos {
  left: 0;
  top: 0;
}

html:not([dir="rtl"]) .rtlPos {
  right: 0;
  top: 0;
}

html[dir="rtl"] .smRtlVRPos {
  left: var(--sema-space-300);
  top: var(--sema-space-500);
}

html:not([dir="rtl"]) .smRtlVRPos {
  right: var(--sema-space-300);
  top: var(--sema-space-500);
}

html[dir="rtl"] .lgRtlVRPos {
  left: var(--sema-space-400);
  top: var(--sema-space-400);
}

html:not([dir="rtl"]) .lgRtlVRPos {
  right: var(--sema-space-400);
  top: var(--sema-space-400);
}
