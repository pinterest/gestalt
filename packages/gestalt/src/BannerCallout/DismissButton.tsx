import classnames from 'classnames';
import styles from '../BannerCallout.css';
import { useDefaultLabelContext } from '../contexts/DefaultLabelProvider';
import InternalIconCompactButton from '../IconButton/InternalIconCompactButton';

type Props = {
  size?: 'sm' | 'lg';
  dismissButton?: {
    accessibilityLabel?: string;
    onDismiss: () => void;
  };
};

export default function DismissButton({ dismissButton, size = 'lg' }: Props) {
  const { accessibilityDismissButtonLabel } = useDefaultLabelContext('BannerCallout');

  return (
    <div
      className={classnames(
        styles.dismissButton,
        size === 'lg' ? styles.lgRtlVRPos : styles.smRtlVRPos,
      )}
    >
      <InternalIconCompactButton
        accessibilityLabel={dismissButton?.accessibilityLabel ?? accessibilityDismissButtonLabel}
        icon="compact-cancel"
        iconColor="darkGray"
        onClick={dismissButton?.onDismiss}
        size="sm"
      />
    </div>
  );
}
