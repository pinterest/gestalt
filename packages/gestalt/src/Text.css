/*
This file was generated by running

  $ /Users/chrislloyd/.nvm/versions/node/v9.2.1/bin/node /Users/chrislloyd/src/gestalt/scripts/text.js

*/

:root {
  --font-size-1: 12px;
  --font-size-2: 14px;
  --font-size-3: 16px;
  --font-size-4: 18px;
  --font-size-5: 21px;
}

.Text {
  composes: antialiased sansSerif letterSpacing from "./Typography.css";
}

.fontSize1 {
  font-size: var(--font-size-1);
}

.fontSize2 {
  font-size: var(--font-size-2);
}

.fontSize3 {
  font-size: var(--font-size-3);
}

.fontSize4 {
  font-size: var(--font-size-4);
}

.fontSize5 {
  font-size: var(--font-size-5);
}

@media (--sm) {
  .smFontSize1 {
    font-size: var(--font-size-1);
  }

  .smFontSize2 {
    font-size: var(--font-size-2);
  }

  .smFontSize3 {
    font-size: var(--font-size-3);
  }

  .smFontSize4 {
    font-size: var(--font-size-4);
  }

  .smFontSize5 {
    font-size: var(--font-size-5);
  }
}

@media (--md) {
  .mdFontSize1 {
    font-size: var(--font-size-1);
  }

  .mdFontSize2 {
    font-size: var(--font-size-2);
  }

  .mdFontSize3 {
    font-size: var(--font-size-3);
  }

  .mdFontSize4 {
    font-size: var(--font-size-4);
  }

  .mdFontSize5 {
    font-size: var(--font-size-5);
  }
}

@media (--lg) {
  .lgFontSize1 {
    font-size: var(--font-size-1);
  }

  .lgFontSize2 {
    font-size: var(--font-size-2);
  }

  .lgFontSize3 {
    font-size: var(--font-size-3);
  }

  .lgFontSize4 {
    font-size: var(--font-size-4);
  }

  .lgFontSize5 {
    font-size: var(--font-size-5);
  }
}
