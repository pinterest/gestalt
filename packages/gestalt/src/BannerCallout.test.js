// @flow strict
import { create } from 'react-test-renderer';
import BannerCallout from './BannerCallout';
import Text from './Text';

describe('<BannerCallout />', () => {
  test('Error BannerCallout', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="error"
        message="Insert a clever error bannercallout message here"
        type="error"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Warning BannerCallout', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="warning"
        message="Insert a clever warning bannercallout message here"
        type="warning"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Info BannerCallout', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        title="A Title"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction with href', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        primaryAction={{
          href: 'pinterest.com',
          label: 'Visit Pinterest',
          accessibilityLabel: '',
          role: 'link',
        }}
        title="A Title"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction without href', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        primaryAction={{ label: 'Visit Pinterest', accessibilityLabel: '', role: 'button' }}
        title="A Title"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + secondaryAction', () => {
    const tree = create(
      <BannerCallout
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        primaryAction={{
          href: 'pinterest.com',
          label: 'Visit Pinterest',
          accessibilityLabel: '',
          role: 'link',
        }}
        secondaryAction={{
          href: 'pinterest.com/help',
          label: 'Learn more',
          accessibilityLabel: 'Learn more: bannercallout messages',
          role: 'link',
        }}
        title="A Title"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + dismissButton', () => {
    const tree = create(
      <BannerCallout
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: () => {},
        }}
        iconAccessibilityLabel="info"
        message="Insert a clever info bannercallout message here"
        primaryAction={{
          href: 'pinterest.com',
          label: 'Visit Pinterest',
          accessibilityLabel: '',
          role: 'link',
        }}
        title="A Title"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('bannercallout with rich text message', () => {
    const tree = create(
      <BannerCallout
        dismissButton={{
          accessibilityLabel: 'Dismiss this banner',
          onDismiss: () => {},
        }}
        iconAccessibilityLabel="Info"
        message={
          <Text inline>
            You have invited{' '}
            <Text inline weight="bold">
              Leaf Media Agency
            </Text>{' '}
            to your business hierarchy. Once they accept, you will be able to manage their business
            account.
          </Text>
        }
        primaryAction={{
          accessibilityLabel: 'Resend invite',
          label: 'Resend invite',
          role: 'button',
        }}
        secondaryAction={{
          accessibilityLabel: 'Cancel invite',
          label: 'Cancel invite',
          role: 'button',
        }}
        title="You've sent an invite"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
