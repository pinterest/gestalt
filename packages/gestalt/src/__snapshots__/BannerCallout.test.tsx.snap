// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerCallout /> Error BannerCallout 1`] = `
<div
  className="box errorWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="error"
          className="error icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever error bannercallout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> Info BannerCallout 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> Warning BannerCallout 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 warningWeak xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="warning"
          className="warning icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever warning bannercallout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> bannercallout with rich text message 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="Info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              You've sent an invite
            </div>
          </div>
          <span
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            You have invited
             
            <span
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Leaf Media Agency
            </span>
             
            to your business hierarchy. Once they accept, you will be able to manage their business account.
          </span>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label="Cancel invite"
          className="button block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button block accessibilityOutline tapTransition lg transparent enabled childrenDiv"
          >
            <div
              className="default alignCenter Text fontSize300 fontWeightSemiBold"
            >
              Cancel invite
            </div>
          </div>
        </button>
      </div>
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label="Resend invite"
          className="button block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button block accessibilityOutline tapTransition lg white enabled childrenDiv"
          >
            <div
              className="default alignCenter Text fontSize300 fontWeightSemiBold"
            >
              Resend invite
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss this banner"
      className="parentButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="button tapTransition enabled"
      >
        <div
          className="pog roundingCircle transparent"
          style={
            Object {
              "height": 48,
              "width": 48,
            }
          }
        >
          <svg
            aria-hidden={true}
            aria-label=""
            className="default icon iconBlock"
            height={16}
            role="img"
            viewBox="0 0 24 24"
            width={16}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`<BannerCallout /> message + title + primaryAction + dismissButton 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="noOutline inheritColor noUnderline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="default alignCenter Text fontSize300 fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss banner"
      className="parentButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="button tapTransition enabled"
      >
        <div
          className="pog roundingCircle transparent"
          style={
            Object {
              "height": 48,
              "width": 48,
            }
          }
        >
          <svg
            aria-hidden={true}
            aria-label=""
            className="default icon iconBlock"
            height={16}
            role="img"
            viewBox="0 0 24 24"
            width={16}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`<BannerCallout /> message + title + primaryAction + secondaryAction 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label="Learn more: bannercallout messages"
          className="noOutline inheritColor noUnderline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg transparent"
          href="pinterest.com/help"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="default alignCenter Text fontSize300 fontWeightSemiBold"
              >
                Learn more
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="noOutline inheritColor noUnderline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="default alignCenter Text fontSize300 fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> message + title + primaryAction with href 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="noOutline inheritColor noUnderline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="default alignCenter Text fontSize300 fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> message + title + primaryAction without href 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label=""
          className="button block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button block accessibilityOutline tapTransition lg white enabled childrenDiv"
          >
            <div
              className="default alignCenter Text fontSize300 fontWeightSemiBold"
            >
              Visit Pinterest
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BannerCallout /> message + title 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="info icon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="default alignStart breakWord Text fontSize400 fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="default alignStart breakWord Text fontSize300 fontWeightNormal"
          >
            Insert a clever info bannercallout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
